retrieval/targetmanager.go

```diff
@@ -33,7 +33,10 @@ import (
 //
 // The TargetProvider does not have to guarantee that an actual change happened.
 // It does guarantee that it sends the new TargetGroup whenever a change happens.
-// On startup it sends all TargetGroups it can see.
+//
+// Sources() is guaranteed to be called before each call to Run(). On a call to Run()
+// implementing types must send a valid target group for each of the sources they
```

Comment:
To clarify, the `Sources()` are just a way to determine which target groups from before a reload not to drop.